controller/controller.go

```diff
@@ -239,8 +246,8 @@ func (c *Controller) delete(obj Object) {
 }
 
 func (c *Controller) propagate(resourceEvents []ResourceEvent) {
-	topology := c.topology.Build(c.cache.List())
-	c.reconcile(LoggerIntoContext(context.TODO(), c.logger), resourceEvents, topology)
+	topology, err := c.topology.Build(c.cache.List())
+	c.reconcile(LoggerIntoContext(context.TODO(), c.logger), resourceEvents, topology, err)
```

Comment:
For the benefit of the user, we can log the error here, although it doesn't avoid propagating to the reconcile call. WDYT?

```suggestion
	topology, err := c.topology.Build(c.cache.List())
	c.logger.Error(err, "error building topology")
	c.reconcile(LoggerIntoContext(context.TODO(), c.logger), resourceEvents, topology, err)
```